code/test_binary_counter.cpp

```diff
@@ -0,0 +1,35 @@
+#include <iostream>
+#include <functional>
+#include <algorithm>
+#include <iterator>
+#include "binary_counter.h"
+
+template<typename T>
+struct min_op {
+  T operator()(const T& a, const T& b) {
+    return std::min(a, b);
+  }
+};
+
+int main() {
+  char letters[] = { 'C', 'H', 'B',  'F', 'I', 'D', 'E', 'G', 'A', 'J' };
```

Comment:
I think an extra space char snuck in here.

Reply:
fixed